static/app/utils/useNavigate.tsx

```diff
@@ -1,50 +1,65 @@
 import {useCallback, useEffect, useRef} from 'react';
+import type {LocationDescriptor} from 'history';
 
-import {useRouteContext} from 'sentry/utils/useRouteContext';
 import {normalizeUrl} from 'sentry/utils/withDomainRequired';
 
+import useRouter from './useRouter';
+
 type NavigateOptions = {
   replace?: boolean;
   state?: any;
 };
 
+interface ReactRouter3Navigate {
+  (to: LocationDescriptor, options?: NavigateOptions): void;
+  (delta: number): void;
+}
+
 /**
  * Returns an imperative method for changing the location. Used by `<Link>`s, but
  * may also be used by other elements to change the location.
  *
  * @see https://reactrouter.com/hooks/use-navigate
  */
 export function useNavigate() {
-  const route = useRouteContext();
-
-  const navigator = route.router;
+  const router = useRouter();
```

Comment:
just checking but the idea here is for all browserHistory/useRouter usages be switched to useNavigate and then once we switch to react router 6 we move off this last usage of `useRouter`?

Reply:
basically yes. There will be a lot of shim code for react-router 6. We'll then have to migrate off of it.